components/km271_wifi/number.py

```diff
@@ -50,8 +56,42 @@
                 cv.Optional(CONF_MIN_VALUE, default=10): cv.float_,
                 cv.Optional(CONF_STEP, default=0.5): cv.positive_float,
             }),
-            
-
+            cv.Optional(CONF_HC1_ROOMT_TARGET_OFFSET): number.NUMBER_SCHEMA.extend({
+                cv.GenerateID(): cv.declare_id(BuderusParamNumber),
+                cv.Optional(CONF_MAX_VALUE, default=5): cv.float_,
+                cv.Optional(CONF_MIN_VALUE, default=-5): cv.float_,
+                cv.Optional(CONF_STEP, default=1): cv.float_,
+            }),
+            cv.Optional(CONF_HC1_FT_TARGET_MAX): number.NUMBER_SCHEMA.extend({
+                cv.GenerateID(): cv.declare_id(BuderusParamNumber),
+                cv.Optional(CONF_MAX_VALUE, default=90): cv.float_,
+                cv.Optional(CONF_MIN_VALUE, default=20): cv.float_,
+                cv.Optional(CONF_STEP, default=1): cv.positive_float,
+            }),
+            cv.Optional(CONF_HC2_DESIGN_TEMP): number.NUMBER_SCHEMA.extend({
+                cv.GenerateID(): cv.declare_id(BuderusParamNumber),
+                cv.Optional(CONF_MAX_VALUE, default=90): cv.float_,
+                cv.Optional(CONF_MIN_VALUE, default=30): cv.float_,
+                cv.Optional(CONF_STEP, default=1): cv.positive_float,
+            }),
+            cv.Optional(CONF_HC2_ROOMT_TARGET_DAY): number.NUMBER_SCHEMA.extend({
+                cv.GenerateID(): cv.declare_id(BuderusParamNumber),
+                cv.Optional(CONF_MAX_VALUE, default=30): cv.float_,
+                cv.Optional(CONF_MIN_VALUE, default=10): cv.float_,
+                cv.Optional(CONF_STEP, default=0.5): cv.positive_float,
+            }),
+            cv.Optional(CONF_HC2_ROOMT_TARGET_OFFSET): number.NUMBER_SCHEMA.extend({
+                cv.GenerateID(): cv.declare_id(BuderusParamNumber),
+                cv.Optional(CONF_MAX_VALUE, default=5): cv.float_,
+                cv.Optional(CONF_MIN_VALUE, default=-5): cv.float_,
+                cv.Optional(CONF_STEP, default=1): cv.float_,
```

Comment:
The parameter can be written in steps of 0.5 to the controller - so I'd suggest allowing this via home assistant, too
```suggestion
                cv.Optional(CONF_STEP, default=0.5): cv.float_,
```